File: packages/alteration/src/main/java/fdit/alteration/core/incident/Target.java
package fdit.alteration.core.incident;

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlText;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Target {

    public static final String TARGET_ALL = "ALL";

    @JacksonXmlProperty(isAttribute = true)
    private String identifier;

    @JacksonXmlText()
    @JsonProperty("value")
    private String content;

    public String getIdentifier() {
        return identifier;
    }

    public void setIdentifier(final String identifier) {
        this.identifier = identifier;
    }

    public String getContent() {
        return content;
    }

    public void setContent(final String content) {
        this.content = content;
    }
}

File: packages/fdit-ts/src/Incident/Target.ts
export class Target {
  public static readonly TARGET_ALL = 'ALL'
  private identifier?: string
  private content?: string

  public getIdentifier(): string | undefined {
    return this.identifier
  }

  public setIdentifier(identifier: string): void {
    this.identifier = identifier
  }

  public getContent(): string | undefined {
    return this.content
  }
  public setContent(content: string): void {
    this.content = content
  }
}
